r/R/array.R

```diff
@@ -131,6 +133,12 @@ Array <- R6Class("Array",
       assert_is(i, "Array")
       call_function("filter", self, i, options = list(keep_na = keep_na))
     },
+    SortIndices = function(descending = FALSE) {
```

Comment:
IMO `SortIndices` feels a little odd as an R6 method. An alternative approach you could take would be like `match_arrow` in compute.R: create an `order_arrow` function that behaves like `order`, just to encapsulate any logic of how to format the inputs etc. Then you can call that in `sort` and `arrange`.

Reply:
I think if `Take` and `Filter` are R6 methods, then it makes sense for `SortIndices` to be an R6 method. In the C++ compute API, these three functions are all defined in the same way (see [`cpp/src/arrow/compute/api_vector.h`](https://github.com/apache/arrow/blob/master/cpp/src/arrow/compute/api_vector.h)).

My preference would be to keep it as an R6 method and create a separate `order_arrow` function the behaves like `order()` if we want that.

Reply:
Fair point that they go together in that header, but that might mean that Take and Filter don't need to be R6 methods either. Historically, Take and Filter required different handling for Array vs. ChunkedArray etc., but now that the `call_function` interface exists, that's not the case, and in fact the C++ interface is functional rather than methods-based in that sense. Anyway, something to think about.

Reply:
`SortIndices` does need a different implementation for `Array` vs. `ChunkedArray` as a matter of fact, because of an oddity of the C++ API

Reply:
Should we change the C++ API? I'm pretty sure every other compute function is called the same way regardless of whether it gets an Array or ChunkedArray.

Reply:
I think there might be a good reason for the implementations to be different in the C++ API, but I'm not sure. See the hidden comment here for context: https://github.com/apache/arrow/commit/589cd096059c87b4e34e670d58691cf8ac21d6b0#commitcomment-48425557

Reply:
ARROW-12042